pkg/ccl/crosscluster/logical/logical_replication_writer_processor.go

```diff
@@ -143,10 +149,30 @@ func newLogicalReplicationWriterProcessor(
 
 	dlqDbExec := flowCtx.Cfg.DB.Executor(isql.WithSessionData(sql.NewInternalSessionData(ctx, flowCtx.Cfg.Settings, "" /* opName */)))
 
+	var numTablesWithSecondaryIndexes int
+	for _, td := range tableDescs {
+		if len(td.NonPrimaryIndexes()) > 0 {
+			numTablesWithSecondaryIndexes++
+		}
+	}
+
 	lrw := &logicalReplicationWriterProcessor{
 		spec: spec,
 		getBatchSize: func() int {
-			if useImplicitTxns.Get(&flowCtx.Cfg.Settings.SV) {
+			// We want to decide whether to use implicit txns or not based on
+			// the schema of the target table. Benchmarking has shown that
```

Comment:
One minor note is that the descriptor we are looking at here is the source table.  However, I think for now it is a fiar assumption that the source and the destination are similar.